seed/migrations/0125_dq_refactor.py

```diff
@@ -0,0 +1,62 @@
+# Generated by Django 2.2.10 on 2020-04-13 23:03
+
+from django.db import migrations, models
+
+
+def forwards(apps, schema_editor):
+    Rule = apps.get_model('seed', 'Rule')  # Total 1226
+    # =========================================================================================================== #
+    # data_type not null ---> 1 ---> text_match null: not_null/require/''                                         #
+    #                    |      |--> test_match not null ---> text_match != '': include                           #
+    #                    |                               |--> text_match == '': not_null/required/''              #
+    #                    |                                                                                        #
+    #                    |--> != 1 ---> min & max null: not_null/required/''                                      #
+    #                              |--> min / max not null: range                                                 #
+    #                                                                                                             #
+    # data_type null -------> min & max null ---> text_match null: not_null/required/''                           #
+    #                |                       |--> text_match not null ---> text_match == '': not_null/required/'' #
+    #                |                                                |--> text_match != '': include              #
+    #                |--> min / max not null ---> range                                                           #
+    # =========================================================================================================== #
+    Rule.objects.filter(data_type=1, text_match=None, not_null=True).update(condition='not_null')
+    Rule.objects.filter(data_type=1, text_match=None, required=True, not_null=False).update(condition='required')
+    Rule.objects.filter(data_type=1, text_match=None, required=False, not_null=False).update(condition='')
+    Rule.objects.filter(data_type=1, text_match='').filter(not_null=True).update(condition='not_null')
+    Rule.objects.filter(data_type=1, text_match='').filter(required=True, not_null=False).update(condition='required')
+    Rule.objects.filter(data_type=1, text_match='').filter(required=False, not_null=False).update(condition='')
```

Comment:
I didn't see records in prod that fulfilled this case, but similar question to the text_match=None version of this query..., is there a point to a Rule meeting this criteria?

Reply:
There are 2 records shown in 20200323 dump. Text_match == '' and text_match is null are deemed as 'not_null' or 'required' if corresponding condition is checked for text type rules. If neither condition is checked, it's left blank.